### PR TITLE
feat(filecoin-proofs): implement rocksdb based kv

### LINKED ISSUE #351: replace filesystem-backed k/v store with rocksdb store
rust-fil-proofs `SectorBuilder` metadata is persisted/loaded from a simple datastore which I hacked together in a day.

### PR DESCRIPTION
**Pull request by dignifiedquire**
Closes #351

### CODE REVIEW

File: filecoin-proofs/src/api/sector_builder/kv_store/mod.rs [hunk 1]
```diff
@@ -15,7 +22,7 @@ mod tests {
     #[test]
     fn test_alpha() {
         let metadata_dir = tempfile::tempdir().unwrap();
-
+        println!("dir: {:?}", metadata_dir);
```
Comment (laser) [inline filecoin-proofs/src/api/sector_builder/kv_store/mod.rs]: Errant `println` here :)

**Comment by laser** [comment]
Looks good. I'm glad that somebody finally got around to doing this.

File: filecoin-proofs/src/api/sector_builder/kv_store/mod.rs (hunk 1 above)
Comment (dignifiedquire) [reply filecoin-proofs/src/api/sector_builder/kv_store/mod.rs]: 😵

**Comment by dignifiedquire** [comment]
@laser one question from my side, this currently does not include any migration work. So existing installs will loose their data when switching to this code. Is that an issue?

File: filecoin-proofs/src/api/sector_builder/kv_store/mod.rs (hunk 1 above)
Comment (dignifiedquire) [reply filecoin-proofs/src/api/sector_builder/kv_store/mod.rs]: Fixed

**Pull request merged**